bagit/create_bag_activity.go

```diff
@@ -0,0 +1,139 @@
+package bagit
+
+import (
+	"context"
+	"fmt"
+	"io/fs"
+	"os"
+	"path/filepath"
+
+	gobagit "github.com/nyudlts/go-bagit"
+	cp "github.com/otiai10/copy"
+	"go.artefactual.dev/tools/temporal"
+)
+
+const CreateBagActivityName = "create-bag-activity"
+
+type Config struct {
+	// Algorithm specifies the hashing algorithm used to generate file
+	// checksums. Valid values are ("md5", "sha1", "sha256", "sha512").
+	// Default: "sha512".
+	Algorithm string
+
+	// DirMode specifies the desired file mode for directories in the created
+	// bag. Default: 0o700.
+	DirMode os.FileMode
+
+	// DirMode specifies the desired file mode for files in the created bag.
+	// Default: 0o600.
+	FileMode os.FileMode
+}
```

Comment:
Still having conflicts about config vs params and passing them on activity registration vs execution, what you said in https://github.com/artefactual-sdps/temporal-activities/pull/13#issuecomment-2123125566. I'm not sure if we should follow this pattern in these activities.

Reply:
Yes, it is a difficult choice.  I think the main considerations in whether to set vars in config or execution params are:

1. If it's a value that a system administrator need to configure for different deployments, then the variable should be available in a config file
2. If we anticipate having to call the same activity multiple times with different values for the variable, then it must be passed as an execution param

In this case I don't know if a system admin will need to change any of these values for different deployments.  The choice of hashing algorithm has trade-offs of CPU time vs. collision resistance, so I can different deployments wanting to make different choices based on system resources and security concerns.  If we assume that the pre-processing worker & processing worker are run by the same user on the same machine then I don't see any need for administrator configuration of the file modes.

I can't see any reason we'd want to call the CreateBagActivity multiple times in preprocessing, and if we did I'd assume that we'd want to use the same hashing algorithm for each call.

So maybe in this case the algorithm should be a config option and the file modes should be execution parameters?

Reply:
> If it's a value that a system administrator need to configure for different deployments, then the variable should be available in a config file

It can be available in the worker config and be passed to the activity as a parameter.

> I can't see any reason we'd want to call the CreateBagActivity multiple times in preprocessing, and if we did I'd assume that we'd want to use the same hashing algorithm for each call.

It's more about setting patterns in this repository, not for this particular activity. There could be a worker running multiple workflows using the same activity; I think the activity could be registered multiple times with different names and configs in that case, just a thought.

Passing configuration at registration increases the likelihood of an error, which is what brought us here. We will give the responsibility to the caller to validate that config before registering, which I'm not saying is a bad decision, but it kinds of defeats the idea of making it configurable from the activity.

Maybe the activity config should implement their own `Validate` function and be run as part of the worker config validation (like we do here: https://github.com/artefactual-sdps/enduro/blob/main/internal/config/config.go#L54-L61).

Reply:
@jraddaoui I like the pattern of having a `config.Validate()` method that is run just after config reading/parsing so we can handle config validation errors there ­­­— I'll implement that for the `bagit.Config`.